src/cgame/cg_ents.cpp

```diff
@@ -1334,9 +1336,20 @@ void CG_AddPacketEntities()
 		CG_AddCEntity( cent );
 
 		done[ num ] = true;
+
+		if ( num > highest )
+		{
+			highest = num;
+		}
 	}
-	
-	for ( unsigned num = 0; num < MAX_GENTITIES; num++ )
+
+	// On frame 0 there is nothing to remove.
+	static unsigned oldHighest = 0;
+
+	// We assume maxCgEntities is never greater than MAX_GENTITIES.
+	unsigned cg_maxEntities = oldHighest + 1;
```

Comment:
You probably didn't want to make a local variable shadowing the global one

Reply:
Done.